src/Components/test/E2ETest/Tests/RoutingTest.cs

```diff
@@ -1773,11 +1773,11 @@ public void NavigationManagerNavigateToAnotherUrlWithHash_NavigatesToPageAndScro
 
         app.FindElement(By.Id("navigate-test2")).Click();
 
-        var currentWindowScrollY = BrowserScrollY;
         var test2VerticalLocation = app.FindElement(By.Id("test2")).Location.Y;
         var currentRelativeUrl = _serverFixture.RootUri.MakeRelativeUri(new Uri(Browser.Url)).ToString();
-        Assert.Equal("subdir/LongPageWithHash2#test2", currentRelativeUrl);
-        Assert.Equal(test2VerticalLocation, currentWindowScrollY);
+        string expectedUrl = "subdir/LongPageWithHash2#test2";
+        WaitAssert.True(Browser, () => expectedUrl == currentRelativeUrl, default, $"Expected {expectedUrl} but got {currentRelativeUrl}");
```

Comment:
I think you can alternatively use `Browser.Equals`, but I'm fine with this approach too.